TicTacToeRunner/Runner.cs

```diff
@@ -7,69 +7,94 @@ namespace TicTacToeRunner
 {
     class Runner
     {
-        public TicTacToe newGame;
-        public UserInterface gameUI;
-        public bool isGameOver = false;
-
+        private TicTacToe newGame;
+        private UserInterface gameUI;
+        private Options options;
+        private bool isGameOver = false;
 
         static void Main(string[] args)
         {
             var runner = new Runner();
             runner.createGame();
         }
 
-
-        public void createGame()
+        private void createGame()
         {
             gameUI = new UserInterface();
             if (gameUI.startNewGame())
             {
-                var options = new Options(gameUI.setMarkers());
-                newGame = new TicTacToe(options.P1_MARKER, options.P2_MARKER);
-
+                var isSinglePlayer = gameUI.isSinglePlayerGame(gameUI.getTypeOfGame());
+                options = new Options(gameUI.setMarkers(), isSinglePlayer);
+                newGame = new TicTacToe(options.P1_MARKER, options.P2_MARKER, options.IS_SINGLE_PLAYER);
                 while (!isGameOver)
                 {
                     playGameLoop();
                 }
             }
         }
 
-
-        public Dictionary<int, string> createPrimitiveBoard()
+        private void playGameLoop()
         {
-            var primitiveBoard = new Dictionary<int, string>();
-            foreach (Space space in newGame.currentBoard.board)
+            var availableSpaces = newGame.currentBoard.getAvailableSpaces();
+            
+            int selectedSpace = gameUI.getValidSpace(availableSpaces, newGame.currentBoard.createDictBoard(), newGame.currentPlayer);
+
+            bool successfulTurn = newGame.turn(selectedSpace);
+            if (successfulTurn)
             {
-                primitiveBoard.Add(space.location, space.marker);
+                if(newGame.playerTwo.GetType() == typeof(ComputerPlayer))

```

Comment:
This type check shouldn't be necessary. I'll add that to the agenda for this Friday so we can review this more in-depth